### PR TITLE
Stylistic changes to currency

### PR DESCRIPTION
**Pull request by taleh007**
add currency codes variable, following DRY principle

### CODE REVIEW

File: currency/currency [hunk 1]
```diff
@@ -33,16 +37,10 @@ httpGet()
...
-  if [[ $1 != "AUD" && $1 != "BGN" && $1 != "BRL" \
-      && $1 != "CAD" && $1 != "CHF" && $1 != "CNY" && $1 != "CZK" && $1 != "DKK" \
-      && $1 != "EUR" && $1 != "GBP" && $1 != "HKD" && $1 != "HRK" && $1 != "HUF" \
-      && $1 != "IDR" && $1 != "ILS" && $1 != "INR" && $1 != "JPY" && $1 != "KRW" \
-      && $1 != "MXN" && $1 != "MYR" && $1 != "NOK" && $1 != "NZD" && $1 != "PHP" \
-      && $1 != "PLN" && $1 != "RON" && $1 != "RUB" && $1 != "SEK" && $1 != "SGD" \
-      && $1 != "THB" && $1 != "TRY" && $1 != "USD" && $1 != "ZAR" ]]; then
-    echo "1"
-  else
+  if [[ " ${currencyCodes[*]} " == *" $(echo -e "${1}" | tr -d '[:space:]') "* ]]; then
```
Comment (alexanderepstein) [inline currency/currency]: Seems like this line can be reduced to ``` "${currencyCodes[*]}" == *"$(echo -n "${1}")"* ```

Comment (alexanderepstein) [inline currency/currency]: Can you explain how the ```*``` around the ```$1``` checks for the entire array expansion? I haven't seen this done before and seems quite efficient.

Comment (taleh007) [inline currency/currency]: Seems like this line can be reduced to "${currencyCodes[*]}" == *$1*

Comment (taleh007) [inline currency/currency]: > Seems like this line can...
I'm agree with you. But in my opinion input "USD RUB" must be invalidated, not like now. And right now I'm trying to implement this. I will do few commits little bit later

Comment (alexanderepstein) [inline currency/currency]: Ah change ```$1``` to ```$@``` and that should fix that case, nice catch

Comment (alexanderepstein) [inline currency/currency]: So  ``` "${currencyCodes[*]}" == *$@* ```

Comment (taleh007) [inline currency/currency]: But not "AUD BGN" ;)
I think `*" $(echo -e "${@}" | tr -d '[:space:]') "*`  is best solution

Comment (alexanderepstein) [inline currency/currency]: Ah i see the logic now of how that is working, is the -e necessary you don't use escapes anywhere?

Comment (alexanderepstein) [inline currency/currency]: And no space is necessary around the quotes, like this :```"${currencyCodes[*]}" == *"$(echo "${@}" | tr -d '[:space:]')"*```

Comment (taleh007) [inline currency/currency]: @alexanderepstein, yup. It's just a habit. I will delete this

Comment (taleh007) [inline currency/currency]: I'm agree. It was necessary in the beginning of my refactoring

**Comment by alexanderepstein** [comment]
Awesome looks good to merge :tada:, thanks for contributing!

**Pull request merged**